### PR TITLE
Use hosted CI runners for building docker images

### PR DESCRIPTION
**Pull request by muellerzr**
# What does this PR do?

Github lowered the cache/space runners can use. As a result our GPU image (which takes ~6-8gb to build) winds up hitting a storage issue when trying to deploy/build the latest nightly versions for our CI.

This PR uses our self-hosted runner to do this instead, and clears any old caches that the cron already in the system may have missed, to help with storage. 

Fixes # (issue)

- Last side-effect of the docker runner/action failing

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/accelerate/blob/main/CONTRIBUTING.md#submitting-a-pull-request-pr),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/accelerate/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/accelerate/tree/main/docs#writing-documentation---specification).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@BenjaminBossan

### CODE REVIEW

File: .github/workflows/build-docker-images-release.yml [hunk 1]
```diff
@@ -21,44 +21,40 @@ jobs:
 
   version-cpu:
     name: "Latest Accelerate CPU [version]"
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, docker-gpu, multi-gpu]
```
Comment (ydshieh) [inline .github/workflows/build-docker-images-release.yml:24]: do we need to run on 2 different types of VM for building the image(s)?

File: .github/workflows/build-docker-images-release.yml [hunk 2]
```diff
@@ -21,44 +21,40 @@ jobs:
...
         with:
-          context: ./docker/accelerate-cpu
+          file: docker/accelerate-cpu/Dockerfile
           push: true
           tags: huggingface/accelerate-cpu:${{needs.get-version.outputs.version}}
 
   version-cuda:
     name: "Latest Accelerate GPU [version]"
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, docker-gpu, multi-gpu]
```
Comment (ydshieh) [inline .github/workflows/build-docker-images-release.yml:44]: same

File: .github/workflows/build-docker-images-release.yml [hunk 3]
```diff
@@ -21,44 +21,40 @@ jobs:
...
       - name: Login to DockerHub
-        uses: docker/login-action@v1
+        uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
 
-      - name: Build and Push GPU
-        uses: docker/build-push-action@v2
+      - name: Build and Push CPU
```
Comment (ydshieh) [inline .github/workflows/build-docker-images-release.yml:55]: This name should be `GPU`?

**Review by ydshieh — APPROVED** [review]
LGTM, but having a few questions :-)

File: .github/workflows/build-docker-images-release.yml (hunk 1 above)
Comment (muellerzr) [reply .github/workflows/build-docker-images-release.yml:24]: We do not, these are just all the tags for this particular machine.

**Comment by muellerzr** [comment]
@BenjaminBossan I did wind up having to use the gh action workflow, simply because some things with self-hosted runners get mucky

File: .github/workflows/build-docker-images-release.yml (hunk 1 above)
Comment (ydshieh) [reply .github/workflows/build-docker-images-release.yml:24]: oh yeah, sorry. Get confused with the matrix thing 😅

Comment (muellerzr) [reply .github/workflows/build-docker-images-release.yml:24]: No worries at all, fully understandable :)

**Review by BenjaminBossan — APPROVED** [review]
Thanks for taking the time to fix this Docker issue too.

> I did wind up having to use the gh action workflow, simply because some things with self-hosted runners get mucky

Too bad. :crossed_fingers: that this time, it keeps on working.

File: .github/workflows/build-docker-images-release.yml [hunk 4]
```diff
@@ -21,40 +21,40 @@ jobs:
...
     name: "Latest Accelerate CPU [version]"
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, docker-gpu, multi-gpu]
     needs: get-version
     steps:
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
-      - name: Check out code
-        uses: actions/checkout@v2
+        uses: docker/setup-buildx-action@v2
```
[the remark this answers is not in the document]
Comment (muellerzr) [reply .github/workflows/build-docker-images-release.yml:28]: It's the CLI extension for docker build-kit: https://github.com/docker/buildx

**Pull request merged**